Python/compile.c

```diff
@@ -2306,31 +2306,40 @@ static int
 compiler_lambda(struct compiler *c, expr_ty e)
 {
     PyCodeObject *co;
-    PyObject *qualname;
-    static identifier name;
+    PyObject *name, *exprstr;
     Py_ssize_t funcflags;
     arguments_ty args = e->v.Lambda.args;
     assert(e->kind == Lambda_kind);
 
+    exprstr = _PyAST_ExprAsUnicode(e);
+    if (!exprstr) {
+        return 0;
+    }
+    name = PyUnicode_FromFormat("<%U>", exprstr);
```

Comment:
Before this, add C equivalent of, say,
    if len(exprstr) > 36: exprstr = exprstr[0:36] + ' ...'